Feature/pb 181 implement new rules for autolayout

SushiRoll53 opened the pull request:
New rules were already implemented, but there was some mistakes on how they were handle.
Now new rules should be correctly represented
also changed positioned & container to better match the rules

Commit:
Added logic to manage custom on components for proper alignment

Commit:
Cleaned size and container generator

Commit:
Sort children before add spacing

Commit:
Verify there is a child

Commit:
Clening code

Commit:
Updated golden file

Commit:
Run dart build

Commit:
Changed AxisSizing to ParentSizing

Commit:
Removed 3 poins fix after zero

Commit:
Simplify if statement

Commit:
Added missing fields

Commit:
Update auto layout golden tests

Commit:
Update styling tests

Commit:
Added constraints field to custom text form field

Commit:
Specify constraints type

Commit:
Do not modify pin elements

Commit:
Set component isolation to default none

Commit:
Undo changes on constraints

Commit:
Corrected misspelled word

Commit:
Cleaned code

Commit:
WIP moving function from generation service to align strategy

Commit:
Integrated logic into positioned aligment

SushiRoll53 committed:
Merge branch 'dev' into feature/pb-180-isolate-the-positional-assumptions

Commit:
Cleaned code

Commit:
Corrected layoutProperty

Commit:
Added a case for non point value when inside a layoutbuilder

Commit:
Cleaned code

Commit:
Added point value properties higher on the container tree

Commit:
Use PBContainer instead of specific container

Commit:
Ran dart builder

Commit:
Made the case less specific

Commit:
Removed pointValue properties, since it was repetitive from fixed size

Commit:
Re arrange if statements to match better the constraints

Commit:
Updated styling tests

Commit:
Updated auto layout tests

ivan-015 reviewed (changes requested):
Some of the rules regarding stacks are still not being interpreted correctly.
A specific example is the following frame:
<img width="1679" alt="image" src="https://user-images.githubusercontent.com/42812006/175825738-24ed5200-ec85-4bdf-a9ea-533792940330.png">

You can see that horizontally, this frame should scale. However, in the code, the following happens:
<img width="938" alt="image" src="https://user-images.githubusercontent.com/42812006/175825782-4219a7e7-af3b-4695-bbc0-182bd3dd1c21.png">

In line `526` and `527` you can see that the Positioned is using fixed horizontal values rather than relative values. In addition, the Container wrapping that Positioned is also using fixed width.

My guess is that the Positioning rule that inherits constraints is happening for some reason

Commit:
Set widgetbook default

Commit:
Replace enum for boolean

Commit:
Added check if the parent layout sizing is inherit

Commit:
Added orca city screen for autolayout with non autolayout frames

ivan-015 commented:
I noticed that when scaling vertically on `CityDisplayViewScreen`, the following happens:

![image](https://user-images.githubusercontent.com/42812006/176053176-751cdc17-ad28-454b-b434-56558350614e.png)

Here's the code:
![image](https://user-images.githubusercontent.com/42812006/176053232-35356cb4-6908-416c-8787-ee8c432968ca.png)

Here's the figma constraints on the frame:
![image](https://user-images.githubusercontent.com/42812006/176053314-7b239f1b-a239-4191-9892-725cd96d5230.png)

You can see that the constraints are PIN TOP and BOTTOM, while the vertical sizing is FIXED. However, we are exporting a `scaling` height. We might need to check, but I believe that if PIN TOP and BOTTOM are true, we should not generate a height at all since the Positioned is already being created.

ivan-015 commented:
LogIn screen issue:

![image](https://user-images.githubusercontent.com/42812006/176056595-aeea3726-b41d-4667-b7c7-33e43ecf2845.png)

As you can see, the login screen does not center.

Here's the code:

ivan-015 commented:
Unique Value Propsition - View More issue:

![image](https://user-images.githubusercontent.com/42812006/176058039-7cd5912c-cc4c-4053-bd0c-ffb5d1e8bc0e.png)

Code (`dev` branch on the left):
![image](https://user-images.githubusercontent.com/42812006/176058553-9df4a87f-fe58-411c-91be-6b114af32997.png)

Figma constraints:

![image](https://user-images.githubusercontent.com/42812006/176058601-4059019a-d4cc-4930-a57d-84ee588e83ed.png)

Commit:
Re structured align strategy if statements

Commit:
Fixed issue that was not considering the fact of a Layout being vertical or horizontal

Commit:
Special case for constraints on bitmaps

Commit:
Added a case if the pins are set on both sides of width or height do not show it

ivan-015 commented:
We are super close. This is the only bug I noticed regarding AutoLayout-related constraints

<img width="1351" alt="image" src="https://user-images.githubusercontent.com/42812006/176250888-dcb0910d-50d7-430b-8071-da85439db62c.png">

In the frame depicted above, the image on the right moves towards the image on the left as the window scrolls down. However, in Figma, they space between them is always constant:
<img width="420" alt="image" src="https://user-images.githubusercontent.com/42812006/176251081-f1fe7789-26b0-4d2a-8995-3646d99d72c1.png">

Commit:
Updated auto layout golden tests

Commit:
Created 2 new tests for auto layout

Commit:
Fixed new tests names

ivan-015 committed:
Merge branch 'dev' into feature/pb-181-implement-new-rules-for-autolayout

ivan-015 committed:
Corrected order of positioning.

Previously, we were creating the positioned, and then modifying the children constraints, leaving the positioned with outdated constraints

Commit:
Change order of generating height then width

Commit:
Added constraint rules for FrameGroup children

Commit:
Updated styling tests

Commit:
Updated auto layout tests

ivan-015 committed:
Merge branch 'feature/pb-181-implement-new-rules-for-autolayout' of github.com:Parabeac/parabeac_core into feature/pb-181-implement-new-rules-for-autolayout

ivan-015 commented on lib/interpret_and_optimize/entities/subclasses/pb_intermediate_node.dart:
```diff
@@ -193,11 +194,30 @@ abstract class PBIntermediateNode
+    var parentConstraints = json['constraints'];
     rawChildren?.forEach((rawChild) {
       if (rawChild != null) {
+        /// Child inherit Parent's constraints if they are fixed
```
These should take place in the Positional Alignment rules, otherwise we are causing Shotgun Surgery

ivan-015 committed:
Fixed positional rules

ivan-015 committed:
Map LayoutSizing depending on layout for constraints replacement

ivan-015 committed:
Make sure to find nearest parent layout

Commit:
Extended center case, if parent is strech child should be fixed, if parent is fixed child should be fill

SushiRoll53 committed:
Merge pull request #679 from Parabeac/fix/positional-rules

Fixed positional rules

Commit:
Update golden tests

ivan-015 reviewed (approved):
LGTM
